lib/screens/search_screen/search_screen.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_spinkit/flutter_spinkit.dart';
+import 'package:top_git_rep/design/app_localization.dart';
+import 'package:top_git_rep/design/app_theme_text.dart';
+import 'package:top_git_rep/design/dimensions.dart';
+import 'package:top_git_rep/screens/repository_screen/repository_screen.dart';
+
+import 'bloc/github_search_bloc.dart';
+import 'bloc/github_search_event.dart';
+import 'bloc/github_search_state.dart';
+
+class SearchScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<GithubSearchBloc, GithubSearchState>(
+        listener: (context, state) {
+      if (state is RepositoriesState) {
+        Navigator.push(
+          context,
+          MaterialPageRoute(
+            builder: (BuildContext context) => RepositoryScreen(
+              repository: state.repository,
+            ),
+          ),
+        );
+      }
+    }, builder: (BuildContext context, GithubSearchState state) {
+      if (state is LanguagesState) {
+        return Column(
+          children: const <Widget>[
+            SearchBar(),
+            SearchBody(),
+          ],
+        );
+      } else if (state is LoadingState) {
+        return const SpinKitCircle(
+          color: Colors.deepPurpleAccent,
+          size: Dimensions.SPIN_KIT_CIRCLE_SIZE_80,
+        );
+      } else {
+        return const Center(
+          child: Text(AppLocalization.ERROR_SEARCH_FORM),
+        );
+      }
+    });
+  }
+}
+
+class SearchBar extends StatefulWidget {
+  const SearchBar({Key? key}) : super(key: key);
+
+  @override
+  State<SearchBar> createState() => SearchBarState();
+}
+
+class SearchBarState extends State<SearchBar> {
```

Comment:
use stateless widget

it's better to use only bloc and stateless widgets as much as possible

stateful + bloc is possible, when you're working with animations
usually stateful widget's are not necesarry